lib/ansible/modules/network/nxos/nxos_ntp.py

```diff
@@ -266,14 +275,21 @@ def config_ntp(delta, existing):
 
     ntp_cmds = []
     if peer_type:
+        if existing:
+            ntp_cmds.append('no ntp {0} {1}'.format(existing.get('peer_type'),
+                                                    existing.get('address')))
         ntp_cmds.append(set_ntp_server_peer(
             peer_type, address, prefer, key_id, vrf_name))
     if source:
         existing_source_type = existing.get('source_type')
         existing_source = existing.get('source')
         if existing_source_type and source_type != existing_source_type:
             ntp_cmds.append('no ntp {0} {1}'.format(existing_source_type, existing_source))
-        ntp_cmds.append('ntp {0} {1}'.format(source_type, source))
+        if source == 'default':
```

Comment:
Same like the above. Check `if existing_source_type and existing_source:` before hitting nested conditions.

Reply:
@trishnaguha Made the changes. Can you please check if this new code works ok on your setup? All test cases passed for me even before this change so I am not able to get into the condition you described.

Reply:
Instead of checking `if existing_source_type and existing_source:` after `source: default`, I think it would make more sense to add this after Line 285.

Reply:
@trishnaguha I am not able to follow what you are asking. I believe we still need to check if existing source and source type are null or not in both cases. This is what I am thinking, see below. If it is not, please let me know. I will make changes after you reply.
```
if source:
        existing_source_type = existing.get('source_type')
        existing_source = existing.get('source')
        if existing_source_type and source_type != existing_source_type:
            if existing_source:
                ntp_cmds.append('no ntp {0} {1}'.format(existing_source_type, existing_source))
        if source == 'default':
            if existing_source_type and existing_source:
                ntp_cmds.append('no ntp {0} {1}'.format(existing_source_type, existing_source))
        else:
            ntp_cmds.append('ntp {0} {1}'.format(source_type, source))
```

Reply:
@trishnaguha Actually after thinking a bit more, I am not sure if this is needed at all. Since source cannot be None if source_type is not None (basically the CLI cannot be run without specifying source), if we just check for source_type, it is good enough. I think my original code is correct for source/source_type case. For peer_type/address case, I agree with you. 
If you disagree, can you just let me know the config you are using, I will try to replicate the problem.

Reply:
@saichint `no ntp None None` is now fixed, but the idempotence test is now failing https://github.com/saichint/ansible/blob/1873b5744b92d5cd23ce4df1f165113f6b340252/test/integration/targets/nxos_ntp/tests/common/sanity.yaml#L34.
I think that is because `peer_type` key still is there in `delta` after configuring once, unlike `source` key deleted after configuration `delta={'address': '1.2.3.4', 'key_id': '32', 'peer_type': 'server', 'prefer': 'enabled', 'vrf_name': 'management'}`, hence calling the function https://github.com/saichint/ansible/blob/1873b5744b92d5cd23ce4df1f165113f6b340252/lib/ansible/modules/network/nxos/nxos_ntp.py#L282